Online-Messaging-App/src/app/home/sidebar/sidebar.component.ts

```diff
@@ -124,4 +133,25 @@ export class SidebarComponent implements OnInit {
         })
 
     }
+
+    joinChannel():void{
+        const dialogConfig = new MatDialogConfig();
+        dialogConfig.disableClose = true;
+        dialogConfig.autoFocus = true;
+        dialogConfig.width = "35%";
+        let dialogRef = this.dialog.open(CreateChannelComponent,dialogConfig);
+        dialogRef.afterClosed().subscribe((result) => {
+            if(result){
+                this.newChannelEvent.emit(result);
+                this.userSubscribedChannels.push(result);
```

Comment:
pull this and line 63 into a common method, and call them in both places

Reply:
done

Reply:
I meant like line 147 - line 154. That whole thing is reapeated twice, and can be ulled into a method